src/project/mod.rs

```diff
@@ -0,0 +1,60 @@
+//! Project management module
+//!
+//! Manages configuration files, paths, and such.
+
+use std::io;
+use std::path::{Path, PathBuf};
+
+/// Configuration macros.
+#[macro_use]
+pub mod config;
+/// Directory and file loading.
+pub mod directory;
+
+pub use self::config::Config;
+
+/// Error related to anything that manages/creates configurations as well as
+/// "workspace"-related things.
+#[derive(Debug)]
+pub enum ProjectError {
+    /// Forward to the `std::io::Error` error.
+    File(io::Error),
+    /// Errors related to serde's parsing of configuration files.
+    Parser(String),
+}
+
+impl From<io::Error> for ProjectError {
+    fn from(e: io::Error) -> ProjectError {
+        ProjectError::File(e)
+    }
+}
+
+impl ProjectError {
+    /// Returns a human friendly error message for the `ProjectError`.
+    pub fn description(&self) -> String {
+        match self {
+            &ProjectError::File(ref err) => err.to_string(),
+            &ProjectError::Parser(ref msg) => msg.clone(),
+        }
+    }
+}
+
+/// Project structure, holds information related to configurations and meta information.
+pub struct Project {
```

Comment:
Yeah, I'm thinking of having it so the method would take in the `specs::World` and add them as resources. The user would still need to register with the project however, similar to how it works with the specs world, since the project would need to retain the configuration type in a closure.
e.g.
```rust
impl Project {
    fn load(world: &mut World) {
        for config_closure in closures {
            config_closure(world);
        }
    }
    fn register<T: Config>() {
        let closure = |world: specs::World| {
            // find file, load into rust structure with serde
            world.add_resources::<T>(config);
        };
        self.closures.push_back(closure);
    }
}
```

Could also have methods for custom named files of configurations in the config folder.